Frameworks/AddressBook/ABAddressBookManagerInternal.mm

```diff
@@ -55,4 +56,30 @@ - (id)init {
     return self;
 }
 
+- (NSArray*)getListOfContacts {
+    __block NSMutableArray* result = nil;
+    dispatch_semaphore_t semaphore = dispatch_semaphore_create(0);
+
+    [self.contactStore findContactsAsyncWithSuccess:^(NSArray* success) {
+        result = [NSMutableArray arrayWithCapacity:[success count]];
+
+        // Copy over the contacts wrapped in _ABContacts.
+        for (int i = 0; i < [success count]; i++) {
+            result[i] = [[_ABContact alloc] initWithContact:success[i]];
+        }
+
+        dispatch_semaphore_signal(semaphore);
+    }
+        failure:^(NSError* failure) {
+            // In the failure case, result will be nil.
+            dispatch_semaphore_signal(semaphore);
+        }];
+
+    // Wait until async method completes.
+    dispatch_semaphore_wait(semaphore, DISPATCH_TIME_FOREVER);
```

Comment:
>dispatch_semaphore_wait [](start = 4, length = 23)

Although this is a common pattern everywhere else in UIKit where we call projection apis, this is recipe for unresponsive apps.  We don't have a solution yet to this yet (@bbowman and I were talking about this today in the context of a different bug) maybe @mnithish @msft-jeyaram can weigh in potential runloop changes to accommodate this sync from async pattern.